.github/workflows/ci-protobuf.yml

```diff
@@ -0,0 +1,47 @@
+# Continuous integration handling for Protocol Buffers
+name: ci-protobuf
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/ci-protobuf.yml
+      - .go-version
+      - go.mod
+      - go.sum
+      - 'tfprotov5/internal/tfplugin5/*'
+      - 'tfprotov6/internal/tfplugin6/*'
+
+permissions:
+  contents: read
```

Comment:
And this doesn't apply to the filesystem where the generation below will be executed, correct?

I looked on [the doc](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) but it wasn't specified, so I assume the filesystem is always accessible for write, but then any attempt to persist the code would fail.

Also, setting only `contents` leaves all other permissions on the default level, right? [I couldn't determine this either](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).

Reply:
`contents` refers to the REST API endpoints the repository. It's not intuitive, and I'm not sure where that's documented, but its why permissions such as `issues: write` is necessary for modifying milestones, since those APIs are related to issues subsystem.

Assigning one of the permissions removes other permissions per [this documentation](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs):

> If you specify the access for any of these scopes, all of those that are not specified are set to `none`.